src/test/java/database/service/service/JsonDatabaseServiceTest.java

```diff
@@ -0,0 +1,210 @@
+        jsonDatabaseService.addNewRecordToTable(firstStudent);
+
+        ProvidedIdDoesNotExistException exception = assertThrows(ProvidedIdDoesNotExistException.class, () ->
+                jsonDatabaseService.updateRecordInTable(secondStudent, 12));
+
+        assertEquals("Entity with provided Id does not exist.", exception.getMessage());
+        assertTrue(jsonDatabaseService.deleteTable(Student.class));
+    }
+
+    @Test
+    public void removeRecordFromTableTest() throws IOException, DatabaseDoesNotExistException {
+        jsonDatabaseService.createTable(Student.class);
+        jsonDatabaseService.addNewRecordToTable(firstStudent);
+        jsonDatabaseService.addNewRecordToTable(secondStudent);
+
+        assertTrue(jsonDatabaseService.removeRecordFromTable(Student.class, 0));
+        assertFalse(jsonDatabaseService.removeRecordFromTable(Student.class, 13));
+
+        assertTrue(jsonDatabaseService.deleteTable(Student.class));
+    }
+
+    @Test
+    public void removeAllRecordsFromTableTest() throws IOException, DatabaseDoesNotExistException {
+        jsonDatabaseService.createTable(Student.class);
+        jsonDatabaseService.addNewRecordToTable(firstStudent);
+        jsonDatabaseService.addNewRecordToTable(secondStudent);
+
+        jsonDatabaseService.removeAllRecordsFromTable(Student.class);
+        Path databasePath = Path.of(jsonDatabaseService.getDatabasePath(Student.class));
+        assertEquals("[]", Files.readString(databasePath));
+
+        assertTrue(jsonDatabaseService.deleteTable(Student.class));
+    }
+
+    @Test
+    public void getByIdTest() throws IOException, ProvidedIdDoesNotExistException, DatabaseDoesNotExistException {
+        jsonDatabaseService.createTable(Student.class);
+        jsonDatabaseService.addNewRecordToTable(firstStudent);
+        jsonDatabaseService.addNewRecordToTable(secondStudent);
+
+        Student receivedStudentById = jsonDatabaseService.getById(Student.class, 1);
+
+        assertEquals(secondStudent, receivedStudentById);
+        assertEquals(1, receivedStudentById.getId());
+
+        assertTrue(jsonDatabaseService.deleteTable(Student.class));
+    }
+
+    @Test
+    public void getByIdWhenIdDoesNotExistTest() throws IOException, DatabaseDoesNotExistException {
+        jsonDatabaseService.createTable(Student.class);
+        jsonDatabaseService.addNewRecordToTable(firstStudent);
+
+        ProvidedIdDoesNotExistException exception = assertThrows(ProvidedIdDoesNotExistException.class, () ->
+                jsonDatabaseService.getById(Student.class, 13));
+        assertEquals("Entity with provided Id does not exist.", exception.getMessage());
+
+        assertTrue(jsonDatabaseService.deleteTable(Student.class));
+    }
+
+    @Test
+    public void getAllRecordsFromTableTest() throws IOException, DatabaseDoesNotExistException {
+        List<Student> students = new ArrayList<>(List.of(firstStudent, secondStudent));
```

Comment:
Why you add one list to another ?

Reply:
Because I want to add several elements in my list at one time.